src/infra/src/short_url/mysql.rs

```diff
@@ -0,0 +1,194 @@
+            );
+        "#;
+        sqlx::query(query).execute(&pool).await?;
+        Ok(())
+    }
+
+    /// Create index for short_urls at short_id and original_url
+    async fn create_table_index(&self) -> Result<()> {
+        create_index("short_id_idx", "short_urls", true, &["short_id"]).await?;
+        create_index("original_url_idx", "short_urls", true, &["original_url"]).await?;
+        Ok(())
+    }
+
+    /// Add a new entry to the short_urls table
+    async fn add(&self, short_id: &str, original_url: &str) -> Result<()> {
+        let pool = CLIENT.clone();
+        let query = r#"
+            INSERT INTO short_urls (original_url, short_id)
+            VALUES (?, ?)
+            ON DUPLICATE KEY UPDATE original_url = original_url;
+        "#;
+
+        sqlx::query(query)
+            .bind(original_url)
+            .bind(short_id)
+            .execute(&pool)
+            .await?;
+        Ok(())
+    }
+
+    /// Remove an entry from the short_urls table
+    async fn remove(&self, short_id: &str) -> Result<()> {
+        let pool = CLIENT.clone();
+        let query = r#"
+            DELETE FROM short_urls
+            WHERE short_id = ?;
+        "#;
+        sqlx::query(query).bind(short_id).execute(&pool).await?;
+        Ok(())
+    }
+
+    /// Get an entry from the short_urls table
+    async fn get(&self, short_id: &str) -> Result<ShortUrlRecord> {
+        let pool = CLIENT.clone();
+        let query = r#"
+            SELECT short_id, original_url
+            FROM short_urls
+            WHERE short_id = ?;
+        "#;
+        let row = sqlx::query_as::<_, ShortUrlRecord>(query)
+            .bind(short_id)
+            .fetch_one(&pool)
+            .await?;
+
+        Ok(row)
+    }
+
+    /// Get an entry from the short_urls table by original_url
+    async fn get_by_original_url(&self, original_url: &str) -> Result<ShortUrlRecord> {
+        let pool = CLIENT.clone();
+        let query = r#"
+            SELECT short_id, original_url
+            FROM short_urls
+            WHERE original_url = ?;
+        "#;
+        let row = sqlx::query_as::<_, ShortUrlRecord>(query)
+            .bind(original_url)
+            .fetch_one(&pool)
+            .await?;
+        Ok(row)
+    }
+
+    /// List all entries from the short_urls table
+    async fn list(&self) -> Result<Vec<ShortUrlRecord>> {
+        let pool = CLIENT.clone();
+        let query = r#"
+            SELECT short_id, original_url
+            FROM short_urls;
+        "#;
+        let rows = sqlx::query_as::<_, ShortUrlRecord>(query)
+            .fetch_all(&pool)
+            .await?;
+
+        Ok(rows)
+    }
+
+    /// Check if an entry exists in the short_urls table
+    async fn contains(&self, short_id: &str) -> Result<bool> {
+        let pool = CLIENT.clone();
+        let query = r#"
+            SELECT EXISTS (
+                SELECT 1
+                FROM short_urls
+                WHERE short_id = ?
+            );
```

Comment:
I am not sure if this subquery is faster than `select 1 from short_urls where xx`.

Reply:
Update this query to not user any sub queries.